core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -159,6 +159,22 @@ public void log(String format, Object... args) {
         }
     }
 
+    /**
+     * This will log, if error log level is enabled , and throw the runtime exception.
+     * @param runtimeException to be thrown. It will do nothing if {@code null} is provided.
+     */
+    public void logAndThrow(RuntimeException runtimeException) {
+        if(runtimeException == null) {
+            return;
+        }
+        // By default we are treating runtime exception as error ?
+        level = ERROR_LEVEL;
+        if (canLogAtLevel(level)) {
```

Comment:
By default log level in ClientLogger is DISABLED_LEVEL.  Here we are saying that for logAndthrow we want to change loglevel to "ERROR_LEVEL" because we want to print it if it is turned on via ConfigurationManager.getConfiguration()